### PR TITLE
Add remove certificate

### PR DESCRIPTION
**Pull request by emanuele-em**
as the old pr but cleaner

### CODE REVIEW

File: src/macos.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if let Some(search_result) = search_result.first() {
+        if let SearchResult::Ref(Reference::Certificate(cert)) = search_result {
+            cert.delete()?;
+        }
+    }
+
+    add_item(add_option)?;
+
+    Command::new("open")
+        .arg("../macos-add-trusted-cert/macos-add-trusted-cert.app")
```
Comment (mhils) [inline src/macos.rs:34]: This is great for local testing, but that won't work on users' machines. We need to

1) Copy the compiled app into the wheel (in build.rs).
2) Add an argument to `add_trusted_cert` to accept a path for the binary.
3) Add a wrapper in `ffi` to determine the path on the user's machine. You can pretty much copy this: https://github.com/mitmproxy/mitmproxy_rs/blob/da1000e004b645d58ac51cd98329ec7d3d711216/ffi/src/server.rs#L282-L293 :)

**Review by mhils — COMMENTED** [review]
This looks super good and clean! There are only a few things we need to address:

1. We shouldn't commit the `macos-add-trusted-cert` binary to Git. Instead we should build it in CI. Or are there any blockers to that?
2. Let's add `macos-add-trust-cert/README.md` to (briefly) explain why we do this dance with a sub-app.
3. `.arg("../macos-add-trusted-cert/macos-add-trusted-cert.app")` won't work, see below.

File: src/macos.rs (hunk 1 above)
Comment (emanuele-em) [reply src/macos.rs:34]: ok, perfect!

File: macos-certificate-truster/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            .first()
+            .unwrap()
+        {
+            TrustSettings::new(Domain::Admin)
+                .set_trust_settings_always(cert)
+                .unwrap();
+        }
+    }
+    #[cfg(not(target_os = "macos"))]
+    println!("Certificate truster is only available on macos");
```
Comment (emanuele-em) [inline macos-certificate-truster/src/main.rs:25]: This to prevent autofix.ci issues, probably I should adapt autofix?
@mhils

Comment (mhils) [reply macos-certificate-truster/src/main.rs:25]: Ah, yes please. You can teach clippy to ignore the specific package. 👍

File: ffi/src/server.rs [hunk 3]
```diff
@@ -14,6 +14,7 @@ use mitmproxy::network::NetworkTask;
 use mitmproxy::packet_sources::windows::WindowsConf;
 
 use mitmproxy::packet_sources::wireguard::WireGuardConf;
+#[allow(unused_imports)]
```
Comment (mhils) [inline ffi/src/server.rs:17]: leftover?

File: macos-certificate-truster/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "macos-certificate-truster"
+license = "LGPL-3.0-or-later"
```
Comment (mhils) [inline macos-certificate-truster/Cargo.toml:3]: Is this an intentional choice? I'd like to keep everything MIT if possible.

**Review by mhils — COMMENTED** [review]
Looks fantastic now!

File: ffi/src/server.rs (hunk 3 above)
Comment (emanuele-em) [reply ffi/src/server.rs:17]: unused_imports is for ipc on macos

File: macos-certificate-truster/Cargo.toml (hunk 4 above)
Comment (emanuele-em) [reply macos-certificate-truster/Cargo.toml:3]: sure! I looked at [windows redirector](https://github.com/mitmproxy/mitmproxy_rs/blob/main/windows-redirector/Cargo.toml#L3) and thought it was the right thing

Comment (mhils) [reply macos-certificate-truster/Cargo.toml]: Ah yes- the Windows redirector is LGPL because WinDivert and https://crates.io/crates/windivert are LGPL.

**Review by mhils — APPROVED** [review]

**Pull request merged**